From 4fabafb699bc4fa91594fce7fd7eaf57c7fcdede Mon Sep 17 00:00:00 2001
From: "Travis A. Everett" <travis.a.everett@gmail.com>
Date: Fri, 9 Dec 2022 17:11:59 -0600
Subject: [PATCH] resholve: fold in python package deps

Protects resholve and dependents from breakages as py27 support is
removed (or rots).
---
 pkgs/development/misc/resholve/default.nix  |  1 +
 pkgs/development/misc/resholve/deps.nix     | 65 ++++++++++++++++++++-
 pkgs/development/misc/resholve/oildev.nix   |  4 +-
 pkgs/development/misc/resholve/resholve.nix |  3 +-
 4 files changed, 69 insertions(+), 4 deletions(-)

diff --git a/pkgs/development/misc/resholve/default.nix b/pkgs/development/misc/resholve/default.nix
index 3e901070a2e2b..34dd36a212434 100644
--- a/pkgs/development/misc/resholve/default.nix
+++ b/pkgs/development/misc/resholve/default.nix
@@ -22,6 +22,7 @@ rec {
   resholve = callPackage ./resholve.nix {
     inherit (source) rSrc version;
     inherit (deps.oil) oildev;
+    inherit (deps) configargparse;
     inherit resholve-utils;
   };
   # funcs to validate and phrase invocations of resholve
diff --git a/pkgs/development/misc/resholve/deps.nix b/pkgs/development/misc/resholve/deps.nix
index 43882c20cf6c3..ef0ebaa670c15 100644
--- a/pkgs/development/misc/resholve/deps.nix
+++ b/pkgs/development/misc/resholve/deps.nix
@@ -1,4 +1,6 @@
-{ callPackage
+{ lib
+, callPackage
+, fetchFromGitHub
 , python27
 , ...
 }:
@@ -15,5 +17,64 @@
 
 rec {
   # binlore = callPackage ./binlore.nix { };
-  oil = callPackage ./oildev.nix { inherit python27; };
+  oil = callPackage ./oildev.nix {
+    inherit python27;
+    inherit six;
+    inherit typing;
+  };
+  configargparse = python27.pkgs.buildPythonPackage rec {
+    pname = "configargparse";
+    version = "1.5.3";
+
+    src = fetchFromGitHub {
+      owner = "bw2";
+      repo = "ConfigArgParse";
+      rev = "v${version}";
+      sha256 = "1dsai4bilkp2biy9swfdx2z0k4akw4lpvx12flmk00r80hzgbglz";
+    };
+
+    doCheck = false;
+
+    pythonImportsCheck = [ "configargparse" ];
+
+    meta = with lib; {
+      description = "A drop-in replacement for argparse";
+      homepage = "https://github.com/bw2/ConfigArgParse";
+      license = licenses.mit;
+    };
+  };
+  six = python27.pkgs.buildPythonPackage rec {
+    pname = "six";
+    version = "1.16.0";
+
+    src = python27.pkgs.fetchPypi {
+      inherit pname version;
+      sha256 = "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926";
+    };
+
+    doCheck = false;
+
+    meta = {
+      description = "A Python 2 and 3 compatibility library";
+      homepage = "https://pypi.python.org/pypi/six/";
+      license = lib.licenses.mit;
+    };
+  };
+  typing = python27.pkgs.buildPythonPackage rec {
+    pname = "typing";
+    version = "3.10.0.0";
+
+    src = python27.pkgs.fetchPypi {
+      inherit pname version;
+      sha256 = "13b4ad211f54ddbf93e5901a9967b1e07720c1d1b78d596ac6a439641aa1b130";
+    };
+
+    doCheck = false;
+
+    meta = with lib; {
+      description = "Backport of typing module to Python versions older than 3.5";
+      homepage = "https://docs.python.org/3/library/typing.html";
+      license = licenses.psfl;
+    };
+  };
 }
diff --git a/pkgs/development/misc/resholve/oildev.nix b/pkgs/development/misc/resholve/oildev.nix
index 854cbb2a00562..d9659513996d9 100644
--- a/pkgs/development/misc/resholve/oildev.nix
+++ b/pkgs/development/misc/resholve/oildev.nix
@@ -13,6 +13,8 @@
 , cmark
 , file
 , glibcLocales
+, six
+, typing
 }:
 
 rec {
@@ -95,7 +97,7 @@ rec {
 
     nativeBuildInputs = [ re2c file makeWrapper ];
 
-    propagatedBuildInputs = with python27.pkgs; [ six typing ];
+    propagatedBuildInputs = [ six typing ];
 
     doCheck = true;
 
diff --git a/pkgs/development/misc/resholve/resholve.nix b/pkgs/development/misc/resholve/resholve.nix
index 6a71962fd4c5b..d0ed5105ae4cd 100644
--- a/pkgs/development/misc/resholve/resholve.nix
+++ b/pkgs/development/misc/resholve/resholve.nix
@@ -6,6 +6,7 @@
 , rSrc
 , version
 , oildev
+, configargparse
 , binlore
 , resholve-utils
 }:
@@ -19,7 +20,7 @@ python27.pkgs.buildPythonApplication {
 
   propagatedBuildInputs = [
     oildev
-    python27.pkgs.configargparse
+    configargparse
   ];
 
   postPatch = ''
